oap-server/analyzer/meter-analyzer/src/main/java/org/apache/skywalking/oap/meter/analyzer/prometheus/rule/Rules.java

```diff
@@ -40,36 +49,71 @@
-        File[] rules;
-        try {
-            rules = ResourceUtils.getPathFiles(path);
-        } catch (FileNotFoundException e) {
-            throw new ModuleStartException("Load fetcher rules failed", e);
-        }
-        return Arrays.stream(rules)
-            .filter(File::isFile)
-            .map(f -> {
-                try (Reader r = new FileReader(f)) {
-                    String fileName = f.getName();
-                    int dotIndex = fileName.lastIndexOf('.');
-                    fileName = (dotIndex == -1) ? fileName : fileName.substring(0, dotIndex);
-                    if (!enabledRules.contains(fileName)) {
-                        return null;
+    public static List<Rule> loadRules(final String path, List<String> enabledRules) throws IOException {
+
+        final Path root = ResourceUtils.getPath(path);
+
+        Map<String, Boolean> formedEnabledRules = enabledRules
+                .stream()
+                .map(rule -> {
+                    rule = rule.trim();
+                    if (rule.startsWith("/")) {
+                        rule = rule.substring(1);
+                    }
+                    if (!rule.endsWith(".yaml") && !rule.endsWith(".yml")) {
+                        return rule + "{.yaml,.yml}";
                     }
-                    Rule rule = new Yaml().loadAs(r, Rule.class);
-                    rule.setName(fileName);
                     return rule;
-                } catch (IOException e) {
-                    LOG.debug("Reading file {} failed", f, e);
-                }
+                })
+                .collect(Collectors.toMap(rule -> rule, $ -> false));
+        List<Rule> rules;
+        try (Stream<Path> stream = Files.walk(root)) {
+            rules = stream
+                    .filter(it -> formedEnabledRules.keySet().stream()
+                                    .anyMatch(rule -> {
+                                        boolean matches = FileSystems.getDefault().getPathMatcher("glob:" + rule)
+                                                .matches(root.relativize(it));
+                                        if (matches) {
+                                            formedEnabledRules.put(rule, true);
+                                        }
+                                        return matches;
+                                    }))
+                    .map(path1 -> getRulesFromFile(root.relativize(path1), path1))
+                    .filter(Objects::nonNull)
+                    .collect(Collectors.toList()) ;
+        }
+
+        if (formedEnabledRules.containsValue(false)) {
+            throw new UnexpectedException("Some configuration files of enabled rules are not found, enabled rules: " + formedEnabledRules.keySet());
+        }
+        return rules;
+    }
+
+    private static Rule getRulesFromFile(Path relative, Path path) {
```

Comment:
```suggestion
    private static Rule getRulesFromFile(String ruleName, Path path) {
```
I think the caller should build the ruleName, rather than hide the logic inside this method with two Paths as parameters.